### PR TITLE
Language support

### PR DESCRIPTION
**Pull request by Oxydeas**
Added language translations for ENG/FR/PT/ES/IT. Fixed themes not working because of a function in util

### CODE REVIEW

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        sp = PreferenceManager.getDefaultSharedPreferences(this);
         setTheme(getCurrentTheme());
+        Configuration config = getBaseContext().getResources().getConfiguration();
+        Locale locale = new Locale(getCurrentLanguage());
+        Locale.setDefault(locale);
+        config.setLocale(locale);
+        getBaseContext().getResources().updateConfiguration(config, getBaseContext().getResources().getDisplayMetrics());
```
Comment (Oxydeas) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java:27]: This function is deprecated. I found no replacement after several hours

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        sp = PreferenceManager.getDefaultSharedPreferences(this);
         setTheme(getCurrentTheme());
+        Configuration config = getBaseContext().getResources().getConfiguration();
+        Locale locale = new Locale(getCurrentLanguage());
+        Locale.setDefault(locale);
+        config.setLocale(locale);
```
Comment (Defibrilator) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java:26]: Question to understand better, here you get the language from the shared prefs l.24, which returns english as default, but if someone has their phone in italian let's say, does it automatically change to italian and if yes where does it do that or does the person chose in the settings menu ?

**Review by Defibrilator — APPROVED** [review]
Thanks for your work, only one question for understanding.

File: app/src/main/res/values-it-rIT/strings.xml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="delete_button">Cancellare l\'abitazione</string>
+    <string name="edit_title">Abitazione</string>
+    <string name="add_user">Aggiungi utente</string>
+    <string name="change_owner">Cambia proprietario</string>
+    <string name="add_user_hint">Indirizzo e-mail valido</string>
+    <string name="invalid_email">L\'e-mail fornita non è valida</string>
+    <string name="duplicate_user">Utente già presente!</string>
+    <string name="not_owner">Non sei il proprietario!</string>
+    <string name="owner_change_success">Il proprietario è stato cambiato con successo</string>
+    <string name="add_user_success">Aggiunto con successo l\'utente</string>
```
Comment (aindreias) [inline app/src/main/res/values-it-rIT/strings.xml:40]: You could say "Utente aggiunto" (line 40) but we can do language edits later & this is definitely just nitpicking

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java (hunk 2 above)
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java:26]: okay I saw your second sprint task which responds to my question : the user choses in the settings.

**Review by aindreias — APPROVED** [review]
Great job and thanks for fixing the preferences!

**Review by Arkadyus — CHANGES_REQUESTED** [review]
There is an issue when having dark mode on the emulator/phone and using the base theme on the app where the app becomes unusable because of the mix of themes : 
image
Another issue is when changing themes, the house selection screen does not change (I think it is the only activity with that bug)
When selecting a different language than english and trying to open the other activities, you go to mainscreen activity instead no matter which one you were trying to access
Also not sure how easy it would be but is it possible to make the theme change the moment we select it ?

**Comment by aindreias** [comment]
> There is an issue when having dark mode on the emulator/phone and using the base theme on the app where the app becomes unusable because of the mix of themes : image Another issue is when changing themes, the house selection screen does not change (I think it is the only activity with that bug) When selecting a different language than english and trying to open the other activities, you go to mainscreen activity instead no matter which one you were trying to access Also not sure how easy it would be but is it possible to make the theme change the moment we select it ?

The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well.

**Comment by Arkadyus** [comment]
> > There is an issue when having dark mode on the emulator/phone and using the base theme on the app where the app becomes unusable because of the mix of themes : image Another issue is when changing themes, the house selection screen does not change (I think it is the only activity with that bug) When selecting a different language than english and trying to open the other activities, you go to mainscreen activity instead no matter which one you were trying to access Also not sure how easy it would be but is it possible to make the theme change the moment we select it ?
> 
> The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well, so I don't think you should request these changes from Luca's PR.

I see, sorry for that part then, ill approve the changes once the language fix is done

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java (hunk 2 above)
Comment (Oxydeas) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/panels/settings/ThemedAppCompatActivity.java:26]: Yes, we are basically fighting the system. This code is executed in every activity and overrides the default language of the phone

**Comment by aindreias** [comment]
> > The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well, so I don't think you should request these changes from Luca's PR.
> 
> I see, sorry for that part then, ill approve the changes once the language fix is done

No problem!

File: app/src/main/res/values-it-rIT/strings.xml (hunk 3 above)
Comment (Oxydeas) [reply app/src/main/res/values-it-rIT/strings.xml:40]: True, my italian is rusty! I will push a fix later today, thanks for the catch. By the way @sansive if you find anything strange is spanish tell me!

**Comment by Oxydeas** [comment]
I will investigate why you stay on mainscreen when changing languages but to be honest my actual code is only a couple of lines to change the language so I have no idea where to even begin. I will update tonight

File: app/src/main/res/values-es-rES/strings.xml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="address_field">"Introducir dirección "</string>
+    <string name="no">No</string>
+    <string name="warning_address">Introduzca una dirección más precisa</string>
+    <string name="address_error">Imposible encontrar coordenadas</string>
+    <string name="address_fill_fields">"Por favor, rellene ambos campos "</string>
+    <string name="address_network_error">No se pudo conectar a la red</string>
+    <string name="debt">Deuda</string>
+    <string name="color_theme">Tema de colores</string>
+    <string name="lang_title">Idioma</string>
+    <string name="password_constraints">"- Debe contener 8 caracteres mínimos - Debe contener una mayúscula y una minúscula - Debe contener un carácter especial - No debe contener espacio - Las dos contraseñas deben ser idénticas "</string>
```
Comment (sansive) [inline app/src/main/res/values-es-rES/strings.xml:131]: "- Debe contener mínimo 8 caracteres - Debe contener una mayúscula y una minúscula - Debe contener un carácter especial - No debe contener espacio - Las dos contraseñas deben ser idénticas "

**Review by sansive — COMMENTED** [review]
Spanish check done! Good work!

**Review by Arkadyus — APPROVED** [review]
Thanks for the fix !

**Comment by Oxydeas** [comment]
> > > The color theme compatibility w/ dark mode is part of my task this week, same for the theme change the moment we select it - Luca was responsible for the languages and kindly fixed a preference issue that affected themes as well, so I don't think you should request these changes from Luca's PR.
> > 
> > 
> > I see, sorry for that part then, ill approve the changes once the language fix is done
> 
> No problem!

Fixed the refresh issue by sending user to the mainscreen if theme or language changes. Fixed navbar problem

File: app/src/main/res/values-es-rES/strings.xml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="house_speakable_text">Seleccionar</string>
+    <string name="info_speakable_text">"Informaciones "</string>
+    <string name="img_house_row_desc">Imagen de la casa actual</string>
+    <string name="home_logo">Logotipo de la casa</string>
+    <string name="title_activity_settings">Configuración</string>
+    <string name="no_household_info">Aún no hay información</string>
+    <string name="edit_household_notes">Editar la descripción</string>
+    <string name="edit_notes_button">"Editar "</string>
+    <string name="info_header">Residentes</string>
+    <string name="add_household">"Criar una casa "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:15]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="add_join_household">"Adjuntar/criar una casa "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:16]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 7]
```diff
@@ -0,0 +1,133 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <string name="discoverButton">"Descubrir "</string>
+    <string name="appDescription">\"Mejor aplicacón de organización domestica\"- los desarolladores</string>
+    <string name="settings_speakable_text">Configuración</string>
+    <string name="house_speakable_text">Seleccionar</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:6]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="info_speakable_text">"Informaciones "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:7]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="img_house_row_desc">Imagen de la casa actual</string>
+    <string name="home_logo">Logotipo de la casa</string>
+    <string name="title_activity_settings">Configuración</string>
+    <string name="no_household_info">Aún no hay información</string>
+    <string name="edit_household_notes">Editar la descripción</string>
+    <string name="edit_notes_button">"Editar "</string>
+    <string name="info_header">Residentes</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:14]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="add_household">"Criar una casa "</string>
+    <string name="add_join_household">"Adjuntar/criar una casa "</string>
+    <string name="add_household_success">Casa añadida!</string>
+    <string name="add_household_failure">Error al añadir</string>
+    <string name="scanQRButton">Escanea el código QR para unirte</string>
+    <string name="latitude_hint">Latitud</string>
+    <string name="longitude_hint">Longitud</string>
+    <string name="leave_house">Salir</string>
+    <string name="house_name">Nombre de la casa</string>
+    <string name="permission_denied">Esta aplicación necessita este permiso para funcionar correctamente</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:24]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="add_user_hint">Dirección de correo electrónico válida</string>
+    <string name="invalid_email">El correo electrónico propocionado no es válido</string>
+    <string name="duplicate_user">El usuario ya está presente</string>
+    <string name="not_owner">Usted no es el proprietario</string>
+    <string name="owner_change_success">Cambió de proprietario con éxito</string>
+    <string name="add_user_success">Añadió el usuario con éxito</string>
+    <string name="remove_user_success">Borró el usuario con éxito</string>
+    <string name="QR_invalid">Este QR no es una casa válida</string>
+    <string name="remove_user">Borrar el usuario</string>
+    <string name="cant_remove_yourself">No podéis retiraros</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:42]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="cant_remove_owner">El proprietario no se puede retirar</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:43]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="remove_calendar_failure">No puede eliminar el calendario</string>
+    <string name="remove_household_success">Casa borrada con éxito</string>
+    <string name="remove_household_failure">"Error al eliminar "</string>
+    <string name="general_header">General</string>
+    <string name="cache_header">Almacenamiento local</string>
+    <string name="signature_title">Su nombre</string>
+    <string name="dark_mode_title">Modo oscuro</string>
+    <string name="attachment_title">Descargar las piezas adjuntas</string>
+    <string name="attachment_summary_on">"Descargar automáticamente las piezas adjuntas "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:52]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="attachment_summary_off">Descaregar manualmente las piezas adjuntas</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:53]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="task_title_hint">Nombre</string>
+    <string name="task_description_hint">Descripción</string>
+    <string name="add_subtask">Añadir una subtarea</string>
+    <string name="task_completion_title">Enhorabuena!</string>
+    <string name="task_completion_desc">Usted terminó una tarea. Bravo!</string>
+    <string name="subtask_completion_desc">Usted terminó una tarea. Bravo!</string>
+    <string name="tap_to_modify">Tocar para modificar</string>
+    <string name="add_remove_assignee">"Añadir/retirar un asignado "</string>
+    <string name="assignees_button">Asignados</string>
+    <string name="no_events">No hay eventos ese día</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:63]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="date">Fecha (dd/MM/aaaa HH:mm)</string>
+    <string name="add">"Añadir "</string>
+    <string name="cancel">Anular</string>
+    <string name="delete">Borrar</string>
+    <string name="edit">Editar</string>
+    <string name="event_editing_title">Editar evento</string>
+    <string name="confirm">"Confirmar "</string>
+    <string name="email_already_used">Correo electrónico ya utilizado</string>
+    <string name="verify_email">"Comprobe su dirección de correo por favor "</string>
+    <string name="sign_out">Desconexión</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:79]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="duration">Tiempo (min)</string>
+    <string name="date">Fecha (dd/MM/aaaa HH:mm)</string>
+    <string name="add">"Añadir "</string>
+    <string name="cancel">Anular</string>
+    <string name="delete">Borrar</string>
+    <string name="edit">Editar</string>
+    <string name="event_editing_title">Editar evento</string>
+    <string name="confirm">"Confirmar "</string>
+    <string name="email_already_used">Correo electrónico ya utilizado</string>
+    <string name="verify_email">"Comprobe su dirección de correo por favor "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:78]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 18]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="sign_out">Desconexión</string>
+    <string name="sign_in_with_email">Conexión por correo electrónico</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:80]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 19]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="sign_up">"Crear una cuenta "</string>
+    <string name="enter_email">Introducir correo electrónico</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:82]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 20]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="enter_password">Entrar contraseña</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:83]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="confirm_password">Confirmar la contraseña</string>
+    <string name="reg_email_auth_failed">Autenticación imposible</string>
+    <string name="log_email_auth_failed">El correo electrónico o la contraseña está mal</string>
+    <string name="email_not_valid">Correo electrónico inválido</string>
+    <string name="password_not_valid">Contraseña inválida</string>
+    <string name="inputs_empty">Los campos no pueden estar vacíos</string>
+    <string name="signout_intent">"Desconexión "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:90]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="attach">Adjuntar</string>
+    <string name="show">Mostrar</string>
+    <string name="remove">Suprimir</string>
+    <string name="three_dots">Tres puntos</string>
+    <string name="event_attachment_content_description">Los accesorios para el evento</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:95]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="name">Nombre</string>
+    <string name="add_item_title">Añadir un elemento à la lista</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:97]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="item_delete_button_description">Suprimir el elemento correspondiente</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:98]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 25]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="add_item_button">Añadir un elemento</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:99]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 26]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="remove_picked_up">Retirar los elementos recuperados</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:100]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 27]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="bs_add_expense">Añadir un gasto</string>
+    <string name="billsharer">Reparto de facturas</string>
+    <string name="expenses">Gastos</string>
+    <string name="balances">Fondos</string>
+    <string name="amount">Importe</string>
+    <string name="payer">Pagador</string>
+    <string name="offline_warning">"Se ha desconectado. "</string>
+    <string name="show_qr_code">Mostrar código QR</string>
+    <string name="show_invite_qr_code">Mostrar código QR de invitación</string>
+    <string name="house_location_title">Localisación de la casa</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:117]: Done

File: app/src/main/res/values-es-rES/strings.xml [hunk 28]
```diff
@@ -0,0 +1,133 @@
...
+    <string name="title_location_permission">Permiso de ubicación</string>
+    <string name="text_location_permission">Necesitamos acceso a su ubicación para acceder a la casa más cercana</string>
+    <string name="go_back_online_button">Volver a conectarse</string>
+    <string name="go_offline_button">Desconectarse</string>
+    <string name="address_field">"Introducir dirección "</string>
```
[the remark this answers is not in the document]
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:122]: Done

File: app/src/main/res/values-es-rES/strings.xml (hunk 4 above)
Comment (Oxydeas) [reply app/src/main/res/values-es-rES/strings.xml:131]: Done

**Review by aindreias — APPROVED** [review]
Thanks for the changes Luca, and for the theme-related fixes as well. Great work!

**Review by sansive — APPROVED** [review]
Thanks for the translations!

**Pull request merged**